municipal_finance/models/aged_creditor.py

```diff
@@ -35,19 +35,20 @@ class Meta:
 
 
 class AgedCreditorItemsV1(AgedCreditorItems):
-    code = models.TextField(primary_key=True)
+    id = models.AutoField(primary_key=True)
+    code = models.TextField()
 
     class Meta:
         db_table = "aged_creditor_items"
         verbose_name_plural = "Aged Creditor Items (v1)"
 
 
 class AgedCreditorFactsV1(AgedCreditorFacts):
-    item_code = models.ForeignKey(
+    item = models.ForeignKey(
         AgedCreditorItemsV1,
         models.DO_NOTHING,
-        db_column="item_code",
     )
+    item_code = models.TextField()
```

Comment:
item_code should get dropped from the facts table. we won't need it any more. We can get it by joining on the dimension table.

Instead of joining the dimension table on item_code https://github.com/OpenUpSA/municipal-data/blob/master/models/aged_creditor.json#L67 we need to change the babbage model to join on the generated column name item_id. I'm not sure you need to actually use an array like this https://github.com/OpenUpSA/municipal-data/blob/master/models/aged_creditor_v2.json#L67 - I think it should detect the related column, but if just changing item_code in the model to item_id doesn't work (facts and aggregate queries return the expected results), then try the way v2 is configured.

Reply:
Changed this to use the array, it didn't detect the related column